pgmpy/models/DynamicBayesianNetwork.py

```diff
@@ -0,0 +1,500 @@
+
+        EXAMPLE:
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> dbn = DBN()
+        >>> dbn.add_nodes_from(['D','G','I','S','L'])
+        >>> dbn.add_edges_from([(('D',0),('G',0)),(('I',0),('G',0)),(('D',0),('D',1)),(('I',0),('I',1)))])
+        >>> dbn.inter_slice(1)
+        [edge_type(start=('D', 0), end=('D', 1)),
+        edge_type(start=('I', 0), end=('I', 1))]
+        """
+        if not isinstance(timeslice, int):
+            raise ValueError('the timeslice is not an integer value')
+        if timeslice < 0:
+            raise ValueError('negative values for a timeslice are not permissible')
+        if timeslice > self.timestate:
+            raise ValueError('the bayesian network does not contain the inter slice edges of this timeslice')
+        return list(filter(lambda x: x.end[1] == timeslice, self.inter_edges))
+
+    def add_dbn_cpds(self, timeslice, *cpds):
+        """
+        Add CPD (Conditional Probability Distribution) to the Bayesian Model.
+
+        Parameters
+        ----------
+        cpds  :  list, set, tuple (array-like)
+            List of cpds (TabularCPD, TreeCPD, RuleCPD, Factor)
+            which will be associated with the model
+            
+        timeslice :integer value. should be ranging from 0
+                   to the number of the slices present in
+                   the bayesian network.
+
+        EXAMPLE
+        -------
+        >>> from pgmpy.models import DynamicBayesianNetwork as DBN
+        >>> from pgmpy.factors.CPD import TabularCPD
+        >>> student = DBN('D','G','I','S','L')
+        >>> grades_cpd = TabularCPD('G', 3, [[0.1,0.1,0.1,0.1,0.1,0.1],
+        ...                                  [0.1,0.1,0.1,0.1,0.1,0.1],
+        ...                                  [0.8,0.8,0.8,0.8,0.8,0.8]],
+        ...                         evidence=['D', 'I'], evidence_card=[2, 3])
+        >>> student.add_dbn_cpds(0, grades_cpd)
+
+        +------+-----------------------+---------------------+
+        |diff: |          easy         |         hard        |
+        +------+------+------+---------+------+------+-------+
+        |intel:| dumb |  avg |  smart  | dumb | avg  | smart |
+        +------+------+------+---------+------+------+-------+
+        |gradeA| 0.1  | 0.1  |   0.1   |  0.1 |  0.1 |   0.1 |
+        +------+------+------+---------+------+------+-------+
+        |gradeB| 0.1  | 0.1  |   0.1   |  0.1 |  0.1 |   0.1 |
+        +------+------+------+---------+------+------+-------+
+        |gradeC| 0.8  | 0.8  |   0.8   |  0.8 |  0.8 |   0.8 |
+        +------+------+------+---------+------+------+-------+
+        """
+        for cpd in cpds:
+            if not isinstance(cpd, (TabularCPD, TreeCPD, RuleCPD)):
+                raise ValueError('Only TabularCPD, TreeCPD or RuleCPD can be'
+                                 ' added.')
+
+            if (cpd.variable, timeslice) not in self.dbn_cpds:
```

Comment:
Don't try to store the CPDs in this fashion. If you will look at the implementation of all the models, you will find that whether it is CPDs or factors, we don't store them in form of a dictionary with key being the variable name. For example in ```BayesianModel```, ```self.cpds``` is a ```list``` and similarly in ```MarkovModel```, ```self.factors``` is a ```list``` as well. Don't assign the CPDs to the variable in this stage. It should be only assigned to a particular variable only when you need so, (for example while doing inference). So I would suggest that you make ```self.cpds``` as ```{}``` with key only being the time-slice and value being list of all CPDs associated with the model in the particular time-slice.

@ankurankan Your suggestions?